cmd/antrea-agent/agent.go

```diff
@@ -132,76 +132,46 @@ func run(o *Options) error {
 	}
 	defer ovsdbConnection.Close()
 
-	egressEnabled := features.DefaultFeatureGate.Enabled(features.Egress)
-	enableAntreaIPAM := features.DefaultFeatureGate.Enabled(features.AntreaIPAM)
-	enableBridgingMode := enableAntreaIPAM && o.config.EnableBridgingMode
-	enableNodePortLocal := features.DefaultFeatureGate.Enabled(features.NodePortLocal) && o.config.NodePortLocal.Enable
-	l7NetworkPolicyEnabled := features.DefaultFeatureGate.Enabled(features.L7NetworkPolicy)
-	enableMulticlusterGW := features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.EnableGateway
-	enableMulticlusterNP := features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.EnableStretchedNetworkPolicy
-
 	// Bridging mode will connect the uplink interface to the OVS bridge.
-	connectUplinkToBridge := enableBridgingMode
-	ovsDatapathType := ovsconfig.OVSDatapathType(o.config.OVSDatapathType)
-	ovsBridgeClient := ovsconfig.NewOVSBridge(o.config.OVSBridge, ovsDatapathType, ovsdbConnection)
+	connectUplinkToBridge := o.config.EnableBridgingMode
+	ovsBridgeClient := ovsconfig.NewOVSBridge(o.config.OVSBridge, o.ovsDatapathType, ovsdbConnection)
 	ovsCtlClient := ovsctl.NewClient(o.config.OVSBridge)
 	ovsBridgeMgmtAddr := ofconfig.GetMgmtAddress(o.config.OVSRunDir, o.config.OVSBridge)
-	multicastEnabled := features.DefaultFeatureGate.Enabled(features.Multicast)
 	ofClient := openflow.NewClient(o.config.OVSBridge, ovsBridgeMgmtAddr,
-		features.DefaultFeatureGate.Enabled(features.AntreaProxy),
-		features.DefaultFeatureGate.Enabled(features.AntreaPolicy),
-		l7NetworkPolicyEnabled,
-		egressEnabled,
-		features.DefaultFeatureGate.Enabled(features.FlowExporter),
+		o.enableAntreaProxy,
```

Comment:
Do you mean just for `openflow.NewClient` or all modules that need to know any `enableX` flag?
If the latter, could it make the dependency less explicit and harder to detect unreasonable dependency.:

Currently, it's easy to know the dependency graph from the entry function, and when we add one `enableX` flag to one module's constructor, we know that it means the module is going to depend on `X`.

When there is an object containing all flags and I suppose it will be injected to all modules that just need a subset of it. We need to figure out the dependency graph from all module's constructors and perphaps every line of its code if it keeps the flag object as a member variable (meaning it might use one flag in member functions too). And although we don't need to change the parameter list in the entry function when new features are added, the dependency is implicitly injected to every module which has the flag object. Every module could surprisingly start consuming one flag in the object without an explicit declaration (the change of constructor).

For example, recently I found the latest AntreaProxy code is consuming Multicluster's state and is injected Multicluster specific logic from the change of the constructor: https://github.com/antrea-io/antrea/pull/4508/files#diff-17152e40cf8f004606f1ad8434ecb7a731f846f01c6a9bd60fb01ca8c8da7f9dR379, while there could be a more generic way to achieve what it tries to do. It would be harder to know that if AntreaProxy has a global flag object and just consume one flag in one of its function somewhere.